thonny/base_file_browser.py

```diff
@@ -703,6 +706,20 @@ def add_middle_menu_items(self, context):
         if self.supports_directories():
             self.menu.add_command(label=tr("New directory") + "...", command=self.mkdir)
 
+        if self.supports_new_file():
+            self.menu.add_command(label=tr("New file"), command=self.create_new_file)
+
+        if self.supports_rename():
+            self.menu.add_command(label=tr("Rename"), command=self.rename_file)
+
+        if self.supports_copypaste():
+            self.menu.add_command(label=tr("Copy"), command=self.copy_files)
+            self.menu.add_command(label=tr("Cut"), command=self.cut_files)
+            if self.copypaste.has_selection():
+                target = self.get_selected_file()
+                if target and not self.copypaste.conflicts(target):
```

Comment:
What do you think of omitting `conflicts`-check here and relying on the check below? I'm not sure the user will always understand what is the problem when the "Paste" command is simply missing.